hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/util/PlatformName.java

```diff
@@ -45,9 +45,23 @@ public class PlatformName {
 
   /**
    * A public static variable to indicate the current java vendor is
-   * IBM java or not.
+   * IBM and the type is Java Technology Edition which provides its
+   * own implementations of many security packages and Cipher suites.
+   * Note that these are not provided in Semeru runtimes:
+   * See https://developer.ibm.com/languages/java/semeru-runtimes/
+   * The class used is present in any supported IBM JTE Runtimes.
    */
-  public static final boolean IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM");
+  public static final boolean IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM") &&
+    hasClass("com.ibm.security.auth.module.JAASLoginModule");
+
+  private static boolean hasClass(String className) {
+    try {
+      Thread.currentThread().getContextClassLoader().loadClass(className);
+      return true;
+    } catch(ClassNotFoundException ignored) {
```

Comment:
can any other exception get raised here? if so, best to log and downgrade to false too

Reply:
Thanks @steveloughran - Would you suggest we are catching the generic Exception instead?

Reply:
yes